pkg/sbom/scanner/scanner.go

```diff
@@ -220,8 +223,8 @@ func (s *Scanner) handleScanRequest(ctx context.Context, r interface{}) {
 // getImageMetadata returns the image metadata if the collector is a container image collector
 // and the metadata is found in the store.
 func (s *Scanner) getImageMetadata(request sbom.ScanRequest) *workloadmeta.ContainerImageMetadata {
-	store := workloadmeta.GetGlobalStore()
-	if store == nil {
+	store, ok := s.wmeta.Get()
```

Comment:
do we not want to lock `s` in this function ?

Reply:
Not really for the reason explains [here](https://github.com/DataDog/datadog-agent/pull/23933#discussion_r1533874782).